src/http/representation/RepresentationMetadata.ts

```diff
@@ -315,6 +363,7 @@ export class RepresentationMetadata {
 
   public set contentType(input) {
     this.set(CONTENT_TYPE_TERM, input);
+    this.setContentTypeParams(input);
```

Comment:
We always want the `CONTENT_TYPE` value to be the string without parameters (for now at least), so if we allow content-type with parameters to be set through this call, we first need to parse the input string and use the base content-type result as value in the first set instead.

And if the `input` is a `ContentType` object this is going to break.

All of the above could perhaps be incorporated in the `setContentTypeParams` function so you have a `setContentType` function instead.

Reply:
I don't quite follow you here. As far as I could see, the ContentType was always stored as the whole string with parameters in the metadata structure. (eg. `text/plain; charset=utf-8`). To not interfere with how all that is working, I call my setter with the same input (ie. my new `setContenTypeParams(input)` method). The input parameters has always only been of time `string` or `undefined`.

The type of my setter is a remnant of a previous attempt to replace the way ContentType was stored internally (but we chose not to do this). So it is in fact the setContentType(input) function that should be edited I think, with only support for string and undefined.

Reply:
> I don't quite follow you here. As far as I could see, the ContentType was always stored as the whole string with parameters in the metadata structure. (eg. `text/plain; charset=utf-8`). 

In theory yes. In practice, if a class ever did this it would break everything related to content-type so I would see it as a bug if a class ever put `text/plain; charset=utf-8` as the content-type metadata.